motr/client_init.c

```diff
@@ -1618,40 +1613,39 @@ int m0_client_init(struct m0_client **m0c_p,
 	/* Init the hash-table for RM contexts */
 	rm_ctx_htable_init(&m0c->m0c_rm_ctxs, M0_RM_HBUCKET_NR);
 
-#ifdef DTM0
-	rc = m0_conf_process2service_get(m0_reqh2confc(&m0c->m0c_reqh),
-					 &m0c->m0c_reqh.rh_fid,
-					 M0_CST_DTM0, &cli_svc_fid);
-	M0_ASSERT(rc == 0);
-
-	if (m0_dtm0_in_ut()) {
-		/* When in UT, m0c_reqh.rh_fid is the same as the
-		 * server process fid. It causes the client to use
-		 * the server-side service fid. Hard-coding of the client
-		 * service fid resolves this problem.
-		 * TODO: When in UT, walk over the list of processes in
-		 * the conf cache and get the service with "volatile"
-		 * property from there. It will help to get rid of the
-		 * hard-coded service fid.
-		 */
-		cli_svc_fid  = M0_FID_INIT(0x7300000000000001, 0x1a);
-	}
-
 #ifndef __KERNEL__
-	(void) m0_dtm__client_service_start(&m0c->m0c_reqh, &cli_svc_fid);
-#endif
-	m0c->m0c_dtms = m0_dtm0_service_find(&m0c->m0c_reqh);
-	M0_ASSERT(m0c->m0c_dtms != NULL);
+	if (ENABLE_DTM0) {
```

Comment:
Why do we need to substitute #ifndef here with runtime check? That makes the resulting binary more heavy with the branch of code that never happen for particular build.